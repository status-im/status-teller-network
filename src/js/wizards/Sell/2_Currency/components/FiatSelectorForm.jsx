import React, {Component, Fragment} from 'react';
import PropTypes from 'prop-types';
import {FormGroup, Label} from 'reactstrap';
import {Typeahead} from 'react-bootstrap-typeahead';
import {withNamespaces} from 'react-i18next';

class FiatSelectorForm extends Component {
  onInputChange = (text) => {
    const symbol = this.props.currencies.find(x => x.label === text);
    if (symbol) {
      this.props.changeCurrency(symbol.id);
    }

  };

  onChange = (items) => {
    if(items.length){
      const item = items[0];
      this.props.changeCurrency(item.id);
    }
  };

  render() {
    const {t, value} = this.props;
    let defaultSelectedValue = [];
    if (value) {
      const currency = this.props.currencies.find(x => x.id === value);
      defaultSelectedValue.push(currency);
    }

    return (
      <Fragment>
        <h2>{t('sellerFiatContainer.title')}</h2>
        <FormGroup>
          <Label className="text-small mt-3 mb-0">Local currency</Label>
          <Typeahead className="mb-3"
            id="fiatSelector"
            onChange={this.onChange}
            options={this.props.currencies}
            placeholder={t("fiatSelectorForm.placeholder")}
            onInputChange={this.onInputChange}
            submitFormOnEnter={true}
            emptyLabel={t("fiatSelectorForm.emptyLabel")}
            defaultSelected={defaultSelectedValue}
          />
        </FormGroup>
      </Fragment>
    );
  }
}

FiatSelectorForm.propTypes = {
  t: PropTypes.func,
  value: PropTypes.string,
  currencies: PropTypes.array,
  changeCurrency: PropTypes.func
};

export default withNamespaces()(FiatSelectorForm);
