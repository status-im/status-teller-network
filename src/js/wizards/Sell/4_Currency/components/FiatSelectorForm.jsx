import React, {Component, Fragment} from 'react';
import PropTypes from 'prop-types';
import {FormGroup} from 'reactstrap';
import {Typeahead} from 'react-bootstrap-typeahead';
import {withNamespaces} from 'react-i18next';

class FiatSelectorForm extends Component {
  onInputChange = (text) => {
    const symbol = this.props.currencies.find(x => x.label === text);
    if (symbol) {
      this.props.changeCurrency(symbol.id);
    }

  };

  onChange = (items) => {
    if(items.length){
      const item = items[0];
      this.props.changeCurrency(item.id);
    }
  };

  render() {
    const {t, value} = this.props;
    let defaultSelectedValue = [];
    if (value) {
      const currency = this.props.currencies.find(x => x.id === value);
      defaultSelectedValue.push(currency);
    }

    return (
      <Fragment>
        <h2>{t('sellerFiatContainer.title')}</h2>
        <FormGroup>
          <Typeahead className="my-3"
            id="fiatSelector"
            onChange={this.onChange}
            options={this.props.currencies}
            placeholder={t("fiatSelectorForm.placeholder")}
            onInputChange={this.onInputChange}
            submitFormOnEnter={true}
            emptyLabel={t("fiatSelectorForm.emptyLabel")}
            defaultSelected={defaultSelectedValue}
          />
          {!this.props.value && <p className="text-muted">{t("fiatSelectorForm.selectValid")}</p>}
        </FormGroup>
      </Fragment>
    );
  }
}

FiatSelectorForm.propTypes = {
  t: PropTypes.func,
  value: PropTypes.string,
  currencies: PropTypes.array,
  changeCurrency: PropTypes.func
};

export default withNamespaces()(FiatSelectorForm);
