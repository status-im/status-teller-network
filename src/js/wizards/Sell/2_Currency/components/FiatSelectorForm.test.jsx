import React from "react";
import { shallow } from "enzyme";

import FiatSelectorForm from "./FiatSelectorForm";

describe('FiatSelectorForm', () => {
  it('should render correctly', () => {
    const component = shallow(<FiatSelectorForm value={""} 
                                                currencies={[]}
                                                changeFiat={() => undefined}/>);
  
    expect(component).toMatchSnapshot();
  });
});
