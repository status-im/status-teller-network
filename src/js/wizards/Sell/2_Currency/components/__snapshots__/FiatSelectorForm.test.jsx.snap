// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiatSelectorForm should render correctly 1`] = `
<Fragment>
  <h2 />
  <FormGroup
    tag="div"
  >
    <Label
      className="text-small mt-3 mb-0"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Local currency
    </Label>
    <TypeaheadContainer(WrappedTypeahead)
      a11yNumResults={[Function]}
      a11yNumSelected={[Function]}
      align="justify"
      allowNew={false}
      autoFocus={false}
      caseSensitive={false}
      className="mb-3"
      clearButton={false}
      defaultInputValue=""
      defaultOpen={false}
      defaultSelected={Array []}
      disabled={false}
      dropup={false}
      emptyLabel=""
      filterBy={Array []}
      flip={false}
      highlightOnlyResult={false}
      id="fiatSelector"
      ignoreDiacritics={true}
      inputProps={Object {}}
      isInvalid={false}
      isLoading={false}
      isValid={false}
      labelKey="label"
      maxResults={100}
      minLength={0}
      multiple={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onInputChange={[Function]}
      onKeyDown={[Function]}
      onPaginate={[Function]}
      options={Array []}
      paginate={true}
      paginationText="Display additional results..."
      placeholder=""
      selectHintOnEnter={false}
      submitFormOnEnter={true}
    />
  </FormGroup>
</Fragment>
`;
